packages/xo-server/src/xapi/mixins/storage.js

```diff
@@ -84,4 +84,28 @@ export default {
     })
     return unhealthyVdis
   },
+
+  async createSr({
+    hostRef,
+    name_label, // eslint-disable-line camelcase
+    name_description = '', // eslint-disable-line camelcase
+    type,
+    device_config = {}, // eslint-disable-line camelcase
+    shared = false,
+  }) {
+    const srRef = await this.call(
+      'SR.create',
+      hostRef,
+      device_config,
+      '0',
```

Comment:
Shouldn't it be an integer?

Reply:
yes, something weird is afoot, it's not been rejected in my tests.

Reply:
All the calls in sr.js use a string too, I guess I copied it from there.
I really need to go to bed now I'll fix it in 8 hours. 

do you want me to fix also the calls in sr.js (the issue is that I can't test them)?

Reply:
XAPI can be really permissive sometimes (especially for integers which must be encoded as strings when using XML-RPC).

But I think it's best to use the correct types.

> I really need to go to bed now I'll fix it in 8 hours.

My comments are never urgent, answer them when you can, don't miss any sleep on them please :slightly_smiling_face: 

> do you want me to fix also the calls in sr.js (the issue is that I can't test them)?

Nope, let's just focus on new code for the moment.

Reply:
fixed 
Do you want me to add a comment "please replace by a call to `xapi.createSr()` at first opportunity" on all the sr creations in sr.js?

Reply:
No, we'll do it write away in another PR.

@HamadaBrest Can you create another PR on top of this one using `xapi.createSr()` in `api/sr.js`?